feat(frontend): implement question type components (#540)

minhle35 opened the pull request:
Overview
This PR implements the five question component types requested in issue #540. These components enhance the application's form capabilities by supporting various response formats.
Key Changes

- Implemented five question component types: ShortAnswer, Dropdown, MultiChoice, MultiSelect, and Ranking
- Created a test route for component demonstration and validation


Testing
- Currently using the test route `/question-components-test` for manual testing of these components. Below are screenshots of the current implementation.

![Screenshot 2025-04-14 at 11 34 37 AM](https://github.com/user-attachments/assets/e8e13280-d8c0-48f5-80c4-e697cc8f31a7)
![Screenshot 2025-04-14 at 11 34 53 AM](https://github.com/user-attachments/assets/614a78da-80a2-4ab8-8809-2432a6e5af56)
![Screenshot 2025-04-14 at 11 35 25 AM](https://github.com/user-attachments/assets/30368eff-c28f-41d5-b441-2fd13fd3a0f9)


----
Fixes per request
1. Short Answer: extend the width and height (Increased width of textarea using cols attribute)
<img width="804" alt="short_answers_fixed" src="https://github.com/user-attachments/assets/3d905676-77ed-405b-b706-8f6c57bd3573" />

2. Dropdown box: add borderline, improve hover effects, and highlight selected options
<img width="937" alt="drop_down_fixed_initial_state" src="https://github.com/user-attachments/assets/4b2ffcc1-1c07-49cb-9d18-5125ed0d1e5f" />
<img width="927" alt="drop_down_after_select" src="https://github.com/user-attachments/assets/e7d08b7a-7ac2-40ed-afc7-4775195f75fa" />

3. Ranking: improve UI by integrating React DnD
<img width="480" alt="ranking_fixed_initial_state" src="https://github.com/user-attachments/assets/0f6b7ab8-682b-43a4-90cd-67fd65e06c84" />
<img width="433" alt="ranking_after_ranked" src="https://github.com/user-attachments/assets/32891325-f252-4773-8268-d8d7d9b6ecf0" />

Gyoumi commented:
Looks mostly good, a few things which could be fixed:

1. Short Answer
- It would be good if the border of the textbox was visible even if not selected, just so it is clear where the textbox is and where it ends
- It shows in your screenshots, but in the browsers I have tested (Chrome, Opera, Vivaldi) the text indicator does not appear when I click into the textbox. It would be good if we can have it be visible and flashing so that users can know when the textbox is active and ready to accept text

2.  Ranking
- Similarly with the textbox, would be good if we could see exactly where and when each draggable element begins ends. 
- It would also be good if the draggable icon was inline with the rest of the element. You can try putting it on the left or right side (or remove it entirely) and see which looks best

minhle35 commented on frontend/src/components/QuestionComponents/MultiChoice/index.tsx:
remove first line

minhle35 commented:
Fixed two components
1. Short answer: Replaced current `textarea` with existing TextArea component
   - Improves UI consistency
   - Adds  focus states and hover effects
   
<img width="959" alt="image" src="https://github.com/user-attachments/assets/23306c60-b777-4907-8f65-f37bce6a9237" />

2. Ranking: Added interactive hover effects for drag-and-drop items

https://github.com/user-attachments/assets/53ce4ac2-7266-48f5-bfc4-2bc1dc03df2f
